pinecone/management_client_test.go

```diff
@@ -0,0 +1,146 @@
+package pinecone
+
+import (
+	"context"
+	"fmt"
+	"github.com/stretchr/testify/suite"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+type ManagementClientTests struct {
+	suite.Suite
+	client  ManagementClient
+	project Project
+	apiKey  APIKeyWithoutSecret
+}
+
+func TestManagementClient(t *testing.T) {
+	suite.Run(t, new(ManagementClientTests))
+}
+
+func (ts *ManagementClientTests) SetupSuite() {
+	apiKey := os.Getenv("ORG_API_KEY")
+	require.NotEmpty(ts.T(), apiKey, "ORG_API_KEY env variable not set")
+
+	client, err := NewManagementClient(NewManagementClientParams{ApiKey: apiKey})
+	if err != nil {
+		ts.FailNow(err.Error())
+	}
+	ts.client = *client
+
+	projects, err := ts.client.ListProjects(context.Background())
+	require.NoError(ts.T(), err, "Failed to list projects for test setup")
+	require.Greater(ts.T(), len(projects), 0, "Projects list should not be empty")
+	ts.project = *projects[0]
+
+	apiKeys, err := ts.client.ListApiKeys(context.Background(), ts.project.Id)
+	require.NoError(ts.T(), err, "Failed to list API keys for test setup")
+	require.NotNil(ts.T(), apiKeys, "API keys in test setup should not be nil")
+	require.Greater(ts.T(), len(apiKeys), 0, "API key list in test setup should not be empty")
+	ts.apiKey = *apiKeys[0]
+}
+
+func (ts *ManagementClientTests) TestListProjects() {
+	projects, err := ts.client.ListProjects(context.Background())
+	require.NoError(ts.T(), err, "Failed to list projects")
+	require.Greater(ts.T(), len(projects), 0, "Projects list should not be empty")
+}
+
```

Comment:
Are all of these tests happy path? Should we add some sad path test cases?

Reply:
Yeah, this is a great callout. The entire test suite could use more robust testing. It's difficult to trigger some of the sad paths since these are hitting live servers. Should be holistically considered before v1.0